<launch>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
    <param name="video_device" value="/dev/video0" />
    <param name="pixel_format" value="yuyv" />
    <param name="image_width" value="1280" />
    <param name="image_height" value="720" />
  </node>
  <!-- Launch the contour_publisher node -->
  <node name="contour_publisher" pkg="robotSelfie" type="main.py" output="screen"/>

  <node name="contours_echo" pkg="rostopic" type="rostopic" args="echo /contours" output="screen" launch-prefix="gnome-terminal --title 'Contours Echo' -x"/>

  <!-- Launch the path_optimizer node -->
  <node name="path_optimizer" pkg="robotSelfie" type="path_optimizer" output="screen" launch-prefix="gnome-terminal --title 'Path Optimizer' -x"/>

  <node name="optimized_path_echo" pkg="rostopic" type="rostopic" args="echo /optimized_path" output="screen" launch-prefix="gnome-terminal --title 'Optimized Path Echo' -x"/>

  <!-- Launch rqt_graph -->
  <node name="rqt_graph" pkg="rqt_graph" type="rqt_graph" output="screen" launch-prefix="gnome-terminal --title 'RQT Graph' -x"/>

  <!-- Launch RViz with the updated configuration file -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find robotSelfie)/rviz/config.rviz"/>

  <!-- Add more nodes as needed -->
</launch>